docs/konnectors-workflow.md

```diff
@@ -0,0 +1,557 @@
+# Konnectors workflow
+
+## TL;DR
+
+![Konnectors workflow](diagrams/konnector-workflow.png)
+
+![Konnectors models](diagrams/konnector-models.png)
+
+
+## Types
+
+### Accounts
+
+`io.cozy.accounts` contains information for an account, including those for
+authentication (technically, they are encrypted by cozy-stack before being
+saved in CouchDB, but we will show them as the konnectors will see them):
+
+```json
+{
+    "auth": {
+        "login": "000000000",
+        "password": "**********"
+    },
+    "folderPath": "/Administratif/Free Mobile",
+    "label": "freemobile",
+    "namePath": "Free Mobile"
+}
+```
+
+Accounts are manipulated through the `/data/` API.
+
+**Note:** you can read more about the [accounts doctype
+here](https://docs.cozy.io/en/cozy-doctypes/docs/io.cozy.accounts/).
+
+### Konnectors
+
+`io.cozy.konnectors` are installed similarly to `io.cozy.apps` (see
+[doc](./konnectors.md))
+
+### Permissions
+
+Like client-side applications, each konnector has an associated
+`io.cozy.permissions`. These permissions are those listed on the manifest of
+the konnectors, and the stack will add a permission for the files on the folder
+choosen by the user.
+
+### Triggers
+
+`io.cozy.triggers` are used to define when konnectors are launched. See
+https://docs.cozy.io/en/cozy-stack/jobs/#post-jobstriggers
+
+
+## Complete flow example
+
+As a user, from the expenses management app, I have a clean flow to configure a
+connector to retrieve my travel expenses
+
+1 - User is in **my-expenses** and clicks on \[configure travels\]
+
+2 - **my-expenses** triggers an intent
+
+```javascript
+cozy.intents.start("CREATE", "io.cozy.konnectors", {
+    category: "transport"
+});
+```
+
+3 - the Store app catch the intent, fetch all available konnectors and let the
+user choose
+
+```http
+GET /registry?... HTTP/1.1
+```
+
+4 - the user chooses the trainline konnector. Its manifest looks like this:
+
+```json
+{
+    "name": "Trainline",
+    "type": "konnector",
+    "slug": "konnector-trainline",
+    "description": "Konnector for trainline . com",
+    "source": "https://github.com/konnectors/trainlines.git@build",
+    "developer": {
+        "name": "XXX",
+        "url": "https://www.xxx.fr"
+    },
+    "version": "3.0.0",
+    "licence": "AGPL-3.0",
+    "fields": {
+        "save_folder": {
```

Comment:
This kind of file field is deprecated. An [actual manifest](https://github.com/konnectors/cozy-konnector-trainline/blob/master/manifest.konnector#L19) today looks like:
```json
  "fields": {
    "login": {
      "type": "text"
    },
    "password": {
      "type": "password"
    },
    "advancedFields": {
      "folderPath": {
        "advanced": true,
        "isRequired": false
      }
    }
  },
```
In konnector manifest v2, supposed to be used now, we are targeting to replace the `advancedFields.folderPath` by specifying a list of doctypes downloaded by the konnector. See the [related documentation](https://github.com/cozy/cozy-home/blob/master/docs/konnector-manifest.md#manifest-details) about the manifest's `doctype` attribute.